test/e2e/framework/service/jig.go

```diff
@@ -175,7 +178,11 @@ func (j *TestJig) CreateServiceWithServicePort(labels map[string]string, namespa
 			Ports:    ports,
 		},
 	}
-	return j.Client.CoreV1().Services(namespace).Create(service)
+	result, err := j.Client.CoreV1().Services(namespace).Create(service)
+	if err == nil {
+		j.sanityCheckService(result, service.Spec.Type)
+	}
+	return result, err
```

Comment:
Hah. In my original patch I removed that function because it was so mismatched, but then I undid that change to keep the PR simple. I'll add it back...